solr/core/src/java/org/apache/solr/cluster/placement/impl/PluginInteractionsImpl.java

```diff
@@ -0,0 +1,190 @@
+/*
```

Comment:
That's frankly a very confusing name and there's little reason for grouping these classes here. Why not put all implementation classes separately into an *.impl package?

(I know, this increases the number of files :) but that's directly the function of the number of defined interfaces.)

Reply:
I clearly wanted to reduce the number of **files** because somehow it became an issue...

All the implementation classes are already in an `*.impl` package.

Reply:
Heh, ok ... I guess there's no way to make everyone happy ;)